### PR TITLE
fastapi 和 unicorn 实现的可热重启bot示例

### PR DESCRIPTION
**Pull request by fangjiyuan**
fastapi 和 unicorn 实现的可热重启bot示例，可以实现web api定制。

### CODE REVIEW

**Comment by wj-Mcat** [comment]
@fangjiyuan  Please sign cla agreement and make CI green.

File: examples/hotreload_bot.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+#设置发送消息函数
+@app.post('/send_msg')
+async def send_msg(item:Item,bot=Depends(bot)):
+    try:
+        await send_report(bot,item.msg_type,item.name,item.contact_id)
+        return 200
+    except Exception as e:
+        log.exception(e)
+        return 404
+async def send_report(bot,str,str2,str3):
```
Comment (lyleshaw) [inline examples/hotreload_bot.py:50]: 建议这里将 `str,str2,str3` 换为其他变量名，体现变量的含义

File: examples/hotreload_bot.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import sys
+#添加路径，需要修改为自己的文件夹
+sys.path.append(r'd:/github/python-wechaty-puppet-itchat/')
```
Comment (lyleshaw) [inline examples/hotreload_bot.py:3]: 这三行可以考虑删去，一般默认在当前目录下执行

File: examples/hotreload_bot.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+@app.on_event("startup")
+async def bot_main():
+    asyncio.create_task(bot.start())
+    return 200
+#设置发送消息函数
+@app.post('/send_msg')
+async def send_msg(item:Item,bot=Depends(bot)):
+    try:
+        await send_report(bot,item.msg_type,item.name,item.contact_id)
+        return 200
```
Comment (lyleshaw) [inline examples/hotreload_bot.py:46]: 接口返回值建议改为如下格式

```json
{
    'code': 100010,             # error code
    'detail': detail,           # show for user
    'fields': fields,           # show for dev, list
    'success': true             # suc or fail
    'data': data,               # data if success else null, be list or dict
}
```

以符合 RESTful 规范，下同

**Review by lyleshaw — CHANGES_REQUESTED** [review]
Thank you for your PR, I have left some comments for it, I hope you can modify according to it to merge this PR～

**Review by lyleshaw — APPROVED** [review]
LGTM

**Comment by huan** [comment]
Thank you very much for your contribution!

## You are welcome to join [Wechaty Contributor Program](https://wechaty.js.org/docs/contributor-program/)

### 1. Join Wechaty Organization

> [](https://github.com/wechaty)You've invited fangjiyuan to Wechaty! They'll be receiving an email shortly. They can also visit https://github.com/wechaty to accept the invitation.

I have invited you to join our Wechaty GitHub Organization, please accept it by following the above message. (See also: [wechaty/PMC#16](https://github.com/wechaty/PMC/issues/16))

### 2. Update Your Wechaty Contributor Profile

1. Please open [Contributor Hall of Fame](https://docs.google.com/spreadsheets/d/1XcDoIczyIclqXP1p90Sz7S0n4Q22xjFzJmjPFlU2g1E/edit#gid=0) and add yourself to the end of the list, so that other contributors will know you better!
1. Please add yourself to our Website [Contributors](https://wechaty.js.org/contributors/) by [creating a PR](https://github.com/wechaty/jekyll/tree/main/jekyll/_contributors) and refer to this PR link as well.

### 3. Join The Contributor Only WeChat Room

We also have a WeChat room for contributors only which can discuss Wechaty at a deeper level, you are welcome to join and if you are interested.

Please add @lijiarui wechat: _ruirui_0914_ and send her this pr link. She will invite you into Wechaty Contributor Room

Cheers!

**Pull request merged**